src/sql/azdata.proposed.d.ts

```diff
@@ -1801,5 +1821,73 @@ declare module 'azdata' {
 
 		export interface Dialog extends LoadingComponentBase {
 		}
+
+		/**
+		 * Opens the error dialog with customization options provided.
+		 * @param options Dialog options to customize error dialog.
+		 * @returns Id of action button clicked by user, e.g. ok, cancel
+		 */
+		export function openCustomErrorDialog(options: IErrorDialogOptions): Thenable<string | undefined>;
+
+		/**
+		 * Provides dialog options to customize modal dialog content and layout
+		 */
+		export interface IErrorDialogOptions {
+			/**
+			 * Severity Level to identify icon of modal dialog.
+			 */
+			severity: MessageLevel;
+			/**
+			 * Title of modal dialog header.
+			 */
+			headerTitle: string;
+			/**
+			 * Message text to show on dialog.
+			 */
+			message: string;
+			/**
+			 * (Optional) Detailed message, e.g stack trace of error.
+			 */
+			messageDetails?: string;
+			/**
+			 * (Optional) List of custom actions to include in modal dialog alongwith a 'Cancel' button.
+			 * If custom 'actions' are not provided, 'OK' and 'Cancel' buttons will be shown by default.
+			 */
+			actions?: IDialogAction[];
+			/**
+			 * (Optional) If provided, instruction text is shown in bold below message.
+			 */
+			instructionText?: string;
+			/**
+			 * (Optional) If provided, appends read more link after instruction text.
+			 */
+			readMoreLink?: string;
+		}
+
+		/**
+		 * An action that will be rendered as a button on the dialog.
+		 */
+		export interface IDialogAction {
+			/**
+			 * Identifier of action.
+			 */
+			id: string;
+			/**
+			 * Label of Action button.
+			 */
+			label: string;
+			/**
+			 * Defines if button styling and focus should be based on primary action.
+			 */
+			isPrimary: boolean;
+			/**
+			 * Defines if dialog should be closed on action event.
+			 */
+			closeDialog: boolean;
+			/**
+			 * Custom class to append to action button.
+			 */
+			styleClass?: string | undefined;
```

Comment:
what is this? I am ok if you want to pass in some css styles, but passing in a class name doesn't seem to be a good design:
1. extension needs to know the implementation details of ADS core
2. the class name is subject to change, and it will be a runtime error.

Reply:
Yeah, this seems like the reason we added the primary/secondary button types for. Buttons in ADS should all be styled the same so letting the extension completely control the style is probably not what we want.

Reply:
ok, will keep it internal